packages/core/src/stateUtils.ts

```diff
@@ -1434,26 +1438,36 @@ export function resolveActionsAndContext<
       action: typeof action === 'string' ? { type: action } : (action as any)
     };
 
-    if (!('resolve' in resolved)) {
+    if (!('resolve' in resolvedAction)) {
       if (actorCtx?.self.status === ActorStatus.Running) {
-        resolved(args);
+        resolvedAction(args);
+        actorCtx?.system?._sendInspectionEvent({
```

Comment:
The ActorActionEvent is very useful for observing side effects performed by the actor. However, there is not enough information here. It is impossible to associate this action with the transition on which it was executed. Now is it possible to glean which actor performed this event. We might need the following:

```
{
  type: '@xstate.action`,
  sessionId: string, // <-- which actor executed this?
  event: AnyEventObject, // <-- as in ActorTransitionEvent.event
}
```

Reply:
Additionally, an indication whether this was an entry, exit, or a transition action would be nice. It would allow for a better visualization in the Inspector.